llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -939,8 +939,9 @@ static void recursivelyDeleteDeadRecipes(VPValue *V) {
   }
 }
 
-/// Try to fold \p R using TargetFolder to a constant. Will succeed and return a
-/// non-nullptr Value for a handled \p Opcode if all \p Operands are constant.
+/// Try to fold \p R using InstSimplifyFolder to a constant. Will succeed and
+/// return a non-nullptr Value for a handled \p Opcode if \p Operands are
+/// foldable.
```

Comment:
Comment at line 997 should be updated, this now doesn't just fold constant, but folds recipes with all live-in operands to other live-ins.

Reply:
You mean `// Constant folding.`? Isn't it still constant-folding, but with a different folder?

Reply:
Yep. My understanding is that the constant folder folds instructions with constants to other constants and InstSimplifyFolder folds instructions with values (including non-constant) to other existing values, so the result is not necessarily a constant, hence the update to live-ins (which in some sense are 'constant' w.r.t. to VPlan, but not llvm::Constant)